infra/src/main/kotlin/fi/fta/geoviite/infra/geocoding/GeocodingService.kt

```diff
@@ -23,101 +21,103 @@ class GeocodingService(
 
     private val logger: Logger = LoggerFactory.getLogger(this::class.java)
 
-    fun getAddressPoints(locationTrackId: DomainId<LocationTrack>, publishType: PublishType): AlignmentAddresses? {
-        check(locationTrackId is IntId) { "Location track must be stored in DB before calculating address points" }
+    fun getAddressPoints(locationTrackId: IntId<LocationTrack>, publicationState: PublishType): AlignmentAddresses? {
         logger.serviceCall(
             "getAddressPoints",
-            "locationTrackId" to locationTrackId, "publishType" to publishType
+            "locationTrackId" to locationTrackId, "publicationState" to publicationState
         )
-        return addressPointsCache.getAddressPointCacheKey(publishType, locationTrackId)
+        return addressPointsCache.getAddressPointCacheKey(publicationState, locationTrackId)
             ?.let(addressPointsCache::getAddressPoints)
     }
 
-    fun getAddressPointsForPublication(
-        locationTrackId: DomainId<LocationTrack>,
-        kmPostIdsToPublish: List<IntId<TrackLayoutKmPost>>
+    fun getAddressPoints(
+        contextKey: GeocodingContextCacheKey,
```

Comment:
Rajapintana rumempi että tänne pitää anta tuo cacheavain, mutta näin tehtynä julkaisussa ei tarvitse hakea cache-avainta samalle ratanumerolle monesti.